projeto-backend/src/main/java/br/unirio/dsw/compartilhador/api/repository/CompartilhamentoRepository.java

```diff
@@ -24,6 +24,6 @@ public interface CompartilhamentoRepository extends JpaRepository<Compartilhamen
 
 	List<Compartilhamento> findByUsuarioIdAndAceito(Long usuarioId, boolean aceito);
 	
-	@Query("SELECT count(*) FROM Compartilhamento c WHERE c.usuario.id = :ownerId AND c.aceito = 0 AND c.rejeitado = 0 AND c.canceladoDono = 0 AND c.canceladoUsuario = 0") 
+	@Query("SELECT count(*) FROM Compartilhamento c WHERE c.usuario.id = "+ ownerId +" AND c.aceito = 0 AND c.rejeitado = 0 AND c.canceladoDono = 0 AND c.canceladoUsuario = 0") 
```

Comment:
```suggestion
1. **Identificação**  
   Vulnerabilidade de SQL Injection na anotação @Query do método `contarCompartilhamentosAbertos` em `CompartilhamentoRepository.java`. O valor de `ownerId` está sendo concatenado diretamente na query JPQL, o que pode permitir a injeção de código malicioso.

2. **Correção**  
```java
@Query("SELECT count(*) FROM Compartilhamento c WHERE c.usuario.id = :ownerId AND c.aceito = 0 AND c.rejeitado = 0 AND c.canceladoDono = 0 AND c.canceladoUsuario = 0")
long contarCompartilhamentosAbertos(@Param("ownerId") Long ownerId);
```

3. **Justificativa**  
A utilização de parâmetros nomeados (":ownerId") garante que o valor seja corretamente escapado pelo framework JPA/Hibernate, prevenindo ataques de SQL Injection.

4. **Referências**  
- [OWASP Top 10: Injection](https://owasp.org/Top10/A01_2021-Broken_Access_Control/)
- [Spring Data JPA - @Query](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.at-query)
```